regulations/views/preamble.py

```diff
@@ -0,0 +1,55 @@
+from django.http import Http404
+from django.template.response import TemplateResponse
+from django.views.generic.base import View
+
+from regulations.generator.api_reader import ApiReader
+from regulations.generator.generator import LayerCreator
+from regulations.generator.html_builder import HTMLBuilder
+
+
+def find_subtree(root, label_parts):
+    """Given a nested tree and a label to look for, find the associated node
+    in the tree. Note that, unlike regulations, preamble trees _always_ encode
+    their exact position in the tree."""
+    cursor = root
+    while cursor and cursor['label'] != label_parts:
+        next_cursor = None
+        for child in cursor['children']:
+            if child['label'] == label_parts[:len(child['label'])]:
+                next_cursor = child
```

Comment:
I'm expecting that labels are unique across siblings, but for clarity, do we want to `break` once we assign `next_cursor`? Also, do we need error handling here? We don't seem to check that we actually retrieved the requested label parts at the end, if I'm reading correctly.